File: src/896. Monotonic Array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */

export function isMonotonic(nums) {
  return isIncreasing(nums) || isDecreasing(nums)
}

function isIncreasing(nums) {
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] < nums[i - 1]) return false
  }
  return true
}

function isDecreasing(nums) {
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] > nums[i - 1]) return false
  }
  return true
}

File: src/896. Monotonic Array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        def isDecreasing(nums: List[int]) -> bool:
            for i in range(1, len(nums)):
                if nums[i - 1] < nums[i]:
                    return False
            return True

        def isIncreasing(nums: List[int]) -> bool:
            for i in range(1, len(nums)):
                if nums[i - 1] > nums[i]:
                    return False
            return True

        return isDecreasing(nums) or isIncreasing(nums)